lib/puppet/defaults.rb

```diff
@@ -939,26 +945,37 @@ def self.default_checksum_types
 EOT
     },
     :digest_algorithm => {
-        :default  => lambda { default_digest_alg },
+        :default  => lambda { default_digest_algorithm },
         :type     => :enum,
-        :values => ["md5", "sha256", "sha384", "sha512", "sha224"],
+        :values   => %w[md5 sha256 sha384 sha512 sha224],
```

Comment:
How is :values used?  Should this also be set with a lambda that depends on Puppet::Util::Platform.fips_enabled? ?

Reply:
Yeah, I was thinking that too, but the issue is that puppet settings munges/validates the setting before hooks are called. So if I remove `md5` from the list when FIPS is enabled, and you try to use `puppet apply --digest_algorithm md5 -e ...`, then you'll get a mediocre message:

```
Error: Could not initialize global default settings: Invalid value 'md5' for parameter digest_algorithm. Allowed values are 'sha256', 'sha384', 'sha512', 'sha224'
```

I wanted to provide a more helpful message, which means I have to add `md5` to the list of values and do additional validation in the hook.

One alternative is to create a new settings type, though I'm not a fan of proliferation of new setting types.

Another option is to remove the hook logic and stick with the generic error message?

Reply:
Can the user see the value of :values?  In what context?  I think what you have now is OK, as long as the user isn't confused by a list of options without clear documentation about the fips dependency to go along with it.

Reply:
With this PR the error message is:

```
Error: Could not initialize global default settings: MD5 digest is prohibited in FIPS mode. Valid values are sha256, sha384, sha512, sha224.
```

Reply:
Hmm, the new message doesn't specify which setting is causing the issue. Perhaps I should just drop that and use the generic message.